api/src/api/organizations_v1/organization_schemas.py

```diff
@@ -55,14 +59,39 @@ class OrganizationDataSchema(Schema):
     )
 
 
-class OrganizationMemberSchema(Schema):
+class RoleDict(TypedDict):
+    """Type definition for role dictionary in EnrichedOrganizationUser."""
+
+    role_id: UUID
+    role_name: str
+    privileges: set[Privilege]
+
+
+@dataclass
+class EnrichedOrganizationUser:
+    """OrganizationUser enriched with computed fields for serialization.
+
+    This dataclass provides a simple data container that Marshmallow can
+    serialize automatically. All transformation logic is handled in the
+    service layer when constructing instances.
+    """
+
+    user_id: UUID
+    email: str | None
+    roles: list[RoleDict]
+    first_name: str | None
+    last_name: str | None
+    is_ebiz_poc: bool
```

Comment:
Anything stopping these from both being dataclasses just for consistency?

Also, since the schema doesn't need to know about this type anymore, could move it over to the service class where it's used.

---

FYI for the future - if we upgrade our APIFlask version, we'll be able to define schemas with Pydantic which could make all of this a lot smoother as we'd just use that.